fuzz/fuzz_targets/fuzz_expr.rs

```diff
@@ -0,0 +1,184 @@
+// This file is part of the uutils coreutils package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+// spell-checker:ignore parens
+
+#![no_main]
+use libfuzzer_sys::fuzz_target;
+use uu_expr::uumain;
+
+use rand::seq::SliceRandom;
+use rand::Rng;
+use std::ffi::OsString;
+
+use libc::{dup, dup2, STDOUT_FILENO};
+use std::process::Command;
+mod fuzz_common;
+use crate::fuzz_common::is_gnu_cmd;
+
+static CMD_PATH: &str = "expr";
+
+fn run_gnu_expr(args: &[OsString]) -> Result<(String, i32), std::io::Error> {
+    is_gnu_cmd(CMD_PATH)?; // Check if it's a GNU implementation
+
+    let mut command = Command::new(CMD_PATH);
+    for arg in args {
+        command.arg(arg);
+    }
+    let output = command.output()?;
+    let exit_code = output.status.code().unwrap_or(-1);
+    if output.status.success() {
+        Ok((
+            String::from_utf8_lossy(&output.stdout).to_string(),
+            exit_code,
+        ))
+    } else {
+        Err(std::io::Error::new(
+            std::io::ErrorKind::Other,
+            format!("GNU expr execution failed with exit code {}", exit_code),
+        ))
+    }
+}
+
+fn generate_random_string(max_length: usize) -> String {
+    let mut rng = rand::thread_rng();
+    let valid_utf8: Vec<char> = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789"
+        .chars()
+        .collect();
+    let invalid_utf8 = [0xC3, 0x28]; // Invalid UTF-8 sequence
+    let mut result = String::new();
+
+    for _ in 0..rng.gen_range(1..=max_length) {
+        if rng.gen_bool(0.9) {
+            let ch = valid_utf8.choose(&mut rng).unwrap();
+            result.push(*ch);
+        } else {
+            let ch = invalid_utf8.choose(&mut rng).unwrap();
+            if let Some(c) = char::from_u32(*ch as u32) {
+                result.push(c);
+            }
+        }
+    }
+
+    result
+}
```

Comment:
This function looks like a candidate for `fuzz_common.rs` as it is used in `fuzz_test.rs`, too.